### PR TITLE
Clinical submission upload api

### PR DESCRIPTION
**Pull request by UmmulkiramR**
The submission upload will be accessible via a rest API instead of GraphQL.
There is also some code refactoring added

### CODE REVIEW

File: src/submission/submission-api.ts [hunk 1]
```diff
@@ -386,4 +433,18 @@ const isValidCreateBody = (req: Request, res: Response): boolean => {
   return true;
 };
 
+const isValidRequestArgs = (programId: string, uploadedFiles: {}): boolean => {
+  if (programId === undefined) {
+    L.debug('programId missing');
+    // ControllerUtils.badRequest(res, `programId is required`);
+    return false;
+  }
+  if (/*uploadedFiles === undefined &&*/ Object.keys(uploadedFiles).length === 0) {
```
Comment (demariadaniel) [inline src/submission/submission-api.ts:442]: Looks like some test code that wasn't removed?

Comment (UmmulkiramR) [reply src/submission/submission-api.ts]: Cleared it. Thanks.

File: src/schemas/clinical-resolvers/uploadClinicalSubmissions.ts [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+    status = 207;
+  }
+
+  const response = await convertClinicalSubmissionDataToGql(programShortName, {
+    submission: uploadResult?.submission,
+    batchErrors: uploadResult?.batchErrors,
+    successful: uploadResult?.successful,
+  });
+
+  return res.status(status).send(response);
```
Comment (demariadaniel) [inline src/schemas/clinical-resolvers/uploadClinicalSubmissions.ts:54]: Code for returning the response should be limited to the API; the resolver should only return the result of `await convertClinicalSubmissionDataToGql`

Comment (UmmulkiramR) [reply src/schemas/clinical-resolvers/uploadClinicalSubmissions.ts:54]: The _uploadClinicalSubmissions_ adds a layer between the request and the code that processes the request. There is stuff to do to unpack the request and pack the response for the caller which can be handled here. I think it is a good idea to keep it this way.

**Comment by UmmulkiramR** [comment]
cancelling as this was covered in [#1085](https://github.com/icgc-argo/argo-clinical/pull/1085)

**Pull request closed without merging**